includes/Core/Modules/Module.php

```diff
@@ -587,6 +588,21 @@ protected function parse_date_range( $range, $multiplier = 1, $offset = 1, $prev
 		$start_date_offset = $offset + $number_of_days - 1;
 		$date_start        = gmdate( 'Y-m-d', strtotime( $start_date_offset . ' days ago' ) );
 
+		// Check the day of the week alignment.
+		$previous_day_of_week  = gmdate( 'w', strtotime( $date_end ) );
+		$yesterday_day_of_week = gmdate( 'w', strtotime( 'yesterday' ) );
```

Comment:
Why do we specifically use `yesterday` here? Should it not be relative to `$date_start` or at least include the `$offset` ?

Reply:
Here we are comparing the day of the week for the last period vs the previous period. 

The periods are the same number of days, so it doesn't matter if we compare the first day or the last day of the periods - they should each align/be the same day of the week. 

The last period **always ends "yesterday"**, so it is easy to compute.

I'll add some inline docs to clarify the logic here.

Reply:
Thanks @adamsilverstein. I find the "last" vs "previous" rather confusing as to which one means the most recent and which means the second most recent 😄  I understand what it means here since it's documented but elsewhere  it isn't as obvious.

My understanding of the `$offset` here was that this was needed because not all services define the period in the same way so that offsetting by 1 day was necessary in some cases and not in others (or by 2 as mentioned in doc block). If the last period always ends "yesterday" for all services, regardless of `$offset` then I am confused what `$offset` is for.

Reply:
That should have read 'the last period for analytics period always ends yesterday'. In any case, the offset would not change the day alignment logic since it would be added to both the end of the last and previous period, cancelling out in the math. So we can safely ignore offset for this section.

Reply:
you are right about offset and how it is used. For the purposes of the day of the week comparison, we can safely ignore it though, since the previous period and the last period would  **both get the same offset**; the offset added to both the previous and past period so it cancels out when calculating if the days of the weeks align.

Reply:
Hmm as far as I can tell, I'm also thinking this should use the `$offset` instead of hard-coding "yesterday". "yesterday" works for Analytics, since for that the `$offset` is 1, but we should respect that parameter. If we wanted to use `$weekday_align` for e.g. Search Console where the `$offset` is 2, this wouldn't work correctly anymore, since the offset right now is only considered for the previous period's `$date_end`, but not the original period's end date because that is hard-coded to "yesterday".

Reply:
@felixarntz - I updated the code here to include the offset in calculations as well as adding test run's for offset = 2 for all cases.